src/tools/tools/process.fs

```diff
@@ -1171,10 +1171,146 @@ type internal ThreadPoolWait internal (id:int) as this =
         member x.Dispose() = 
             CleanUp.Current.CleanUpAll()
             GC.SuppressFinalize(x)
-            
-
-            
-
+          
+/// ThreadPoolWait.RegisterWaitForSingleObject, with capability:
+/// 1. To remove a particular wait handle. 
+/// 2. To show all wait handles queued in the system, 
+/// and 3. To show long continuation (not normal). 
+type internal ThreadPoolWait() = 
+    static let rwhToRemove = ConcurrentQueue<_>()
+    static let rwhLookup = ConcurrentDictionary<_,_>()
+    /// Maximum wait for the RegisteredWaitHandle to show up
+    static member val private MaxSpinMs = 0 with get, set
+    static member val WarningWaitInMs = 10 with get, set
+    static member val ActiveThreadPools = ConcurrentDictionary<_,_>() with get
+    static member val WaitHandleCollection = ConcurrentDictionary<_,ConcurrentDictionary<_,_>>() with get
+    static member RegisterThreadPool (name:string) = 
+        Logger.LogF( LogLevel.MediumVerbose, ( fun _ -> sprintf "ThreadPoolWait, register wait pool %s ............" name ))
+        ThreadPoolWait.ActiveThreadPools.Item( name ) <- DateTime.UtcNow
+    static member UnregisterThreadPool (name:string) = 
+        Logger.LogF( LogLevel.MildVerbose, ( fun _ -> sprintf "ThreadPoolWait, unregister wait pool %s ............" name ))
```

Comment:
The original thought is to track a pool of handles related to a particular job. But with the job lifecycle object, it may not be necessary, as we can simply register handle wakeup through the CancellationToken with related to the job. I decide to remove both RegisterThreadPool  and UnregisterThreadPool.